02week/ticTacToe.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+const assert = require('assert');
+const readline = require('readline');
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+let board = [
+  [' ', ' ', ' '],
+  [' ', ' ', ' '],
+  [' ', ' ', ' ']
+];
+
+let playerTurn = 'X';
+
+function printBoard() {
+  console.log('   0  1  2');
+  console.log('0 ' + board[0].join(' | '));
+  console.log('  ---------');
+  console.log('1 ' + board[1].join(' | '));
+  console.log('  ---------');
+  console.log('2 ' + board[2].join(' | '));
+}
+
+//my code below
+
+function horizontalWin() {
+  for(board[i] = 0; i < 3; i++) {
+    if (board[i] === ['X', 'X', 'X'] || ['O','O','O']) {
+        return true;
+    } 
+  }
+//written out as if 
+//  if ([0] === ['X', 'X', 'X'] || ['O','O','O']) {
+//    return "Winner!"
+//  }
+ // else if ([1] === ['X', 'X', 'X'] || ['O','O','O']) {
+//    return "Winner!"
+//  }
+  //else if ([2] === ['X', 'X', 'X'] || ['O','O','O']) {
+    //return "Winner!"
+ // }
+}
+
+function verticalWin() {
+  for(board[i][0] = 0; i < 3 ; i++) {
+    if (board[i][0] === 'X'){
+      return true;
+    } else if (board[i][0] === 'O'){
+      return true;
+    }
+  }
+
+  for(board[i][1] = 0; i < 3 ; i++) {
+    if (board[i][1] === 'X'){
+      return true;
+    } else if (board[i][1] === 'O'){
+      return true;
+    }
+  }
+
+  for(board[i][2] = 0; i < 3 ; i++) {
+    if (board[i][2] === 'X'){
+      return true;
+    } else if (board[i][2] === 'O'){
+      return true;
+    }
+  }
+
+function diagonalWin() {
+  if ([2][0] && [1][1] && [0][2] === 'X') {
+    return true;
+  }
+  else if ([2][0] && [1][1] && [0][2] === '0') {
+    return true;
+  }
+
+  for([i][i] = 0; i < 3; i++) {
+    if([i][i] === ['X']) {
+        return true;
+    } 
+    else if ([i][i] === ['O']) {
+       return true;
+    }
+  }
+}
+
+function checkForWin() {
```

Comment:
This will work, but not for the reason you think it will.

You cannot combine checks with just an or in between. You would need to evaluate each condition separetly